src/utilities.js

```diff
@@ -69,68 +53,79 @@ const filterObjectsFromSave = function (save, list) {
 module.exports.filterObjectsFromSave = filterObjectsFromSave;
 
 /**
- *
- * @param {string} type type of validation that is requested
- * @param {float|number} min minimum value a "number" type that can take
- * @param {float|number} max maximum value a "number" type that can take
- * @param selected item value to be checked
- * @param previous default value if item is undefined&NaN
- * @returns {string|*} valid result
+ * Returns if object is number or not
+ * @param {string}n String to be checked
+ * @returns {boolean}
  */
-const getValidData = function (type, min, max, selected, previous) {
-    switch (type) {
-        case "number":
-            return validateNumber(parseFloat(selected), min, max, previous);
-        case "text":
-            if (selected) {
-                return selected.toString();
-            } else if (previous) {
-                return previous.toString();
-            } else {
-                return "";
-            }
-        default:
-            return selected;
-    }
+const isNumeric = function (n) {
+    return !isNaN(parseFloat(n)) && isFinite(n);
 };
 
-module.exports.getValidData = getValidData;
+module.exports.isNumeric = isNumeric;
 
 /**
  * Returns a valid number within constrains
  * @param selected value to be checked
  * @param min minimum value allowed for numbers
  * @param max maximum value allowed for numbers
- * @param previous last valid entry
  * @returns {number|float} valid number
  */
-const validateNumber = function (selected, min, max, previous) {
-    if (isNaN(selected)) {
-        if (previous !== undefined) {
-            if (!isNaN(parseFloat(previous))) {
-                return validateNumber(parseFloat(previous), min, max, undefined);
-            } else {
-                return 0;
-            }
+const clampNumber = function (selected, min, max) {
+    if (selected < min) {
+        return min;
+    } else if (selected > max) {
+        return max;
+    } else {
+        return selected;
+    }
+};
+
+module.exports.clampNumber = clampNumber;
+
+
+/**
+ * Validator for number fields
+ * @param {*}num value to be checked
+ * @param {number}min minimum value allowed for numbers
+ * @param {number}max maximum value allowed for numbers
+ * @param {number|string}defaultValue default value to be passed
+ * @returns {number} valid number
+ */
+const getValidNumber = function (num, min, max, defaultValue = 0) {
+    if (!isNumeric(num)) {
+        if (!isNumeric(defaultValue)) {
+            num = 0;
         } else {
-            if (min !== undefined) {
-                return min;
-            } else {
-                return 0;
-            }
+            return getValidNumber(defaultValue, min, max);
```

Comment:
no need recursive call
`num = defaultValue;`

Reply:
I added this in case default value provided to function is not a valid number. Replacing it fails following unit test:
expect(Utilities.getValidNumber("ABC",undefined,undefined, "ABC")).toBe(0);

Reply:
I should check how the function was used,
defaultValue need check undefined? if it was runtime value,
but then  no min/max checks for defaultValue.
what does the function do for min/max undefined case?

and still no need recursive call, the review have to check it not go infinite loop or
that take an unnecessary risk. flat code is better.